src/cache/event.rs

```diff
@@ -97,9 +97,10 @@ impl CacheUpdate for ChannelDeleteEvent {
         };
 
         // Remove the cached messages for the channel.
-        cache.messages.remove(&self.channel.id());
-
-        None
+        cache
+            .messages
+            .remove(&self.channel.id())
+            .map(|(_, messages)| messages.values().cloned().collect())
```

Comment:
To avoid cloning the messages redundantly, you can change this to either:
```suggestion
            .map(|(_, messages)| messages.into_iter().map(|(_, m)| m).collect())
```
or, since we are bumping the MSRV to 1.63, to:
```suggestion
            .map(|(_, messages)| messages.into_values().collect())
```
(`HashMap::into_values` was added in 1.54)